wagtail/admin/templates/wagtailadmin/shared/header.html

```diff
@@ -23,26 +23,30 @@
     <div class="row">
         <div class="left">
             <div class="col">
-                <h1 class="w-header__title" id="header-title">
-                    {% if icon %}
-                        {% icon classname="w-header__glyph" name=icon %}
-                    {% elif avatar %}
-                        <div class="w-header__glyph avatar large"><img src="{{ avatar }}" alt="" /></div>
-                    {% endif %}
-                    {{ title }}{% if subtitle %} <span class="w-header__subtitle">{{ subtitle }}</span>{% endif %}
-                </h1>
+                {% block h1 %}
+                    <h1 class="w-header__title" id="header-title">
```

Comment:
Maybe we should change this approach, if devs want to completely override the whole h1 content we just let them do that with a variable. This will not allow them to use block.super but if devs need more than a title/icon/subtitle there - then they likely want to override the entire thing.

something like this in `shared/header`

```html
            <div class="col">
                {% if h1 %}
                    {{ h1 }}
                {% else %}
                    <h1 class="w-header__title" id="header-title">
                        {% if icon %}
                            {% icon classname="w-header__glyph" name=icon %}
                        {% elif avatar %}
                            <div class="w-header__glyph avatar large"><img src="{{ avatar }}" alt="" /></div>
                        {% endif %}
                        {{ title }}{% if subtitle %} <span class="w-header__subtitle">{{ subtitle }}</span>{% endif %}
                    </h1>
                {% endif %}
                {% if description %}<div class="w-header__description">{{ description }}</div>{% endif %}
            </div>
```

Then in `modeladmin/header`
```html
{% block content %}
    {% block header %}
        {% fragment as h1_fragment %}{% block h1 %}{% endblock %}{% endfragment %}
....
```

Not sure on this approach but hopefully a start.